packages/ui/src/InteractiveList.tsx

```diff
@@ -0,0 +1,207 @@
+import { IconButton } from './IconButton'
+import { Plus, X } from 'react-feather'
+import styles from './InteractiveList.module.scss'
+import { useUID } from 'react-uid'
+import { FieldArrayRenderProps, FieldValidator, FormikHelpers } from 'formik'
+import { Error, errorId } from './Error'
+import { ExtractKeysOfValueType } from './utils/types'
+
+export type InteractiveListExtraProps = {
+  children?: React.ReactNode | React.ReactNode[]
+  inputControlRef?: Ref<InteractiveListInputRef>
+} & Pick<TextFieldExtraProps<'text'>, 'label' | 'helperText' | 'inputIcon' | 'type' | 'placeholder'>
+
+export type InteractiveListCoreProps<V> = {
+  name: ExtractKeysOfValueType<V, string[]>
+  value: string[]
+  onBlur?: (e: FocusEvent<HTMLTextAreaElement>) => void
+  error?: string
+  validate?: FieldValidator
+  arrayHelpers: FieldArrayRenderProps
+  onError: (str: string) => void
+  validateForm: FormikHelpers<V>['validateForm']
+} & InteractiveListExtraProps
+
+export type InteractiveListProps<V> = InteractiveListExtraProps & InteractiveListCoreProps<V>
+
+export type InteractiveListInputRef = { setValue: (value: string) => void }
+
+export type InteractiveListItemProps = {
+  arrayHelpers: FieldArrayRenderProps
+  content: string
+  error?: string
+  idx: number
+}
+
+export const InteractiveListItem = ({ arrayHelpers, content, error, idx }: InteractiveListItemProps) => {
+  const id = useUID()
+  const errorProps = error
+    ? {
+        'aria-invalid': true,
+        'aria-errormessage': errorId(id),
+      }
+    : {}
+
+  return (
+    <>
+      <div className={styles.content}>
+        <span id={id} {...errorProps}>
+          {content}
+        </span>{' '}
+        <IconButton
+          kind="transparent"
+          ariaLabel="Remove Item"
+          icon={X}
+          size="small"
+          onClick={() => {
+            arrayHelpers.remove(idx)
+          }}
+        />
+      </div>
+      {error && <Error error={error} inputId={id} />}
+    </>
+  )
+}
+
+/**
+ * This component is meant to be used only from InteractiveListFormik
+ * @private
+ */
+const InteractiveList = <V,>({
+  label,
+  inputControlRef,
+  children,
+  error,
+  name,
+  value,
+  helperText,
+  inputIcon,
+  onError,
+  type,
+  validate,
+  arrayHelpers,
+  validateForm,
+}: InteractiveListProps<V>) => {
+  const id = useUID()
+  const [inputValue, setValue] = useState<string>('')
+  // We want to revalidate on change only after un-successful insert
+  const [touched, setTouched] = useState(false)
+
+  const normalizedValue = useMemo(() => inputValue.trim(), [inputValue])
+
+  useImperativeHandle(inputControlRef, () => ({
+    setValue: (value: string) => {
+      setValue(value)
+    },
+  }))
+
+  useEffect(() => {
+    if (touched) {
+      void getValidationError().then((validateError) => {
+        if (validateError) {
+          onError(validateError)
+        }
+      })
+    }
+  }, [touched, inputValue])
```

Comment:
@fantkolja I think this would be valid point in `useCallback`. In `useEffect` I think it's perfectly normal to define when a hook should be run like this, and those pointers doesn't need to be used inside a function directly. Our ESlint rules says so as well, they will throw an error like you mention just for `useCallback`.

Regarding `getValidationError`, it will get the correct value. Like in [React's documentation ](https://reactjs.org/docs/hooks-effect.html#tip-optimizing-performance-by-skipping-effects) they don't include `ChatAPI` in a dependencies, that is the case for `useCallback`

Reply:
@fantkolja I've added those dependencies, it shouldn't hurt anything, the pointers will never change in our case, and we do have an eslint warning for that.

Reply:
I believe the ChatAPI in the example above is taken from the outer scope and btw they have left a note about including all the deps.
![image](https://user-images.githubusercontent.com/13898415/111627755-13472180-87f8-11eb-9b51-fae711cae60b.png)

I think everything will work if you update the `useEffect()` a bit to prevent looping
`  useEffect(() => {
    if (inputTouched) {
      void getValidationError().then((validationError) => {
        // prevent looping
        if (validationError && validationError !== meta.error) {
          onError(validationError)
        }
      })
    }
  }, [inputTouched, inputValue, getValidationError, onError])`
  
And btw with this code snippet the error message is not removed after submit

Reply:
@fantkolja this impl, gives me an infinite loop, even if I wrap getValidationError with `useCallback`